pkg/controller/core/clusterqueue_controller.go

```diff
@@ -562,21 +562,24 @@ func (r *ClusterQueueReconciler) getWorkloadsStatus(cq *kueue.ClusterQueue) *kue
 	if !r.isVisibilityEnabled() {
 		return nil
 	}
-	if r.shouldUpdatePendingWorkloadStatus(cq) {
+	if cq.Status.PendingWorkloadsStatus == nil {
 		return &kueue.ClusterQueuePendingWorkloadsStatus{
```

Comment:
I this case we should return directly ` r.qManager.GetSnapshot(cq.Name)`. Otherwise, there is one interim update that sets empty Status, just with time, which is later replaced. This can be a follow up.

Reply:
I think this is addressed now. Feel free to mark resolved.